docs/plans/2026-07-30-formalize-spec-driven-development.md

```
@@ -0,0 +1,1190 @@
+
+```bash
+sed -n '15,17p;168,171p' docs/specs/2026-04-14-session-manager-v1-design.md
+```
+
+Copy these lines verbatim into the new subsection in Step 2 — do not retype from memory or summarize.
+
+**Bug found during execution, resolved by widening scope rather than narrowing it:** the sed range above spans two of the four citing lines' targets (`:L15`, `:L16`) plus a *superset* of the other two (`:L168-169`, `:L171` — but `:L171`'s cited phrase, "Multi-window session tracking," actually lives at `:L170`; see the D3 correction note in §2 and the off-by-one this step's own range happens to route around by grabbing 168-171 as one block). Reading Step 1's range in isolation and this step's phrasing ("the verbatim passages") both under-specify what "content" means relative to D3(a)'s "fold its content in" and the file-structure table's "absorbs the historical content" framing. Resolved (confirmed via `advisor`): fold in the **entire** historical file, verbatim, as a blockquote — not just the four cited passages — so the delete-after-fold in Step 4 is lossless per this project's own extract-before-deletion standard, and so the off-by-one above is moot rather than silently perpetuated in a partial quote.
+
+- [x] **Step 2: Insert a new subsection into the foundational spec**
+
+Added `### 1.6 Historical record: the v1 session-manager design` immediately after §1.5 ("Acknowledged boundaries of the problem being solved"), containing a one-sentence frame ("Folded in from `2026-04-14-session-manager-v1-design.md` per #84 (D3); quoted verbatim, not restated") followed by the **entire** original file, verbatim, as a `> `-prefixed blockquote (see Step 1's note on why the scope widened from "the four passages" to "the whole file"). Verified byte-identical to `git show HEAD:docs/specs/2026-04-14-session-manager-v1-design.md` (pre-deletion) via a scripted diff, not eyeballing — the file contains em-dashes, `·`, `⚡`, and box-drawing characters that a visual verbatim check would not reliably catch.
```

markdownlint-cli2 (0.23.1)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)